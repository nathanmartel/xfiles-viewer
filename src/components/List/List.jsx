import React from 'react';
import PropTypes from 'prop-types';

const List = ({ monsters }) => {
  
  const monsterList = monsters.map((monster, index) => (
    <li key={index}>
      <a href={`./monster/${name}`}>
        <img src={monster.image} alt={monster.name} />
      </a>
    </li>
  ));

  return (
    <ul>
      {monsterList}
    </ul>
  );
};

List.propTypes = {
  monsters: PropTypes.arrayOf(PropTypes.shape({ 
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }))
};

export default List;
