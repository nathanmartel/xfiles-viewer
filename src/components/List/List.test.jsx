import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const monsters = [{
      name: 'Lorem Ipsum',
      image: 'https://placekitten.com/200/200',
    }, {
      name: 'Dolor Sitar',
      image: 'https://placekitten.com/200/200',
    }];
    const wrapper = shallow(<List monsters={monsters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
