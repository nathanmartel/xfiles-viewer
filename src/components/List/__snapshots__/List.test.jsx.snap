// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <li
    key="0"
  >
    <a
      href="./monster/"
    >
      <img
        alt="Lorem Ipsum"
        src="https://placekitten.com/200/200"
      />
    </a>
  </li>
  <li
    key="1"
  >
    <a
      href="./monster/"
    >
      <img
        alt="Dolor Sitar"
        src="https://placekitten.com/200/200"
      />
    </a>
  </li>
</ul>
`;
